internal/logs/workload.go

```diff
@@ -0,0 +1,199 @@
+package logs
+
+import (
+	"context"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/project-flotta/flotta-device-worker/internal/workload"
+	"github.com/project-flotta/flotta-device-worker/internal/workload/podman"
+	"github.com/project-flotta/flotta-operator/models"
+)
+
+type transportLog struct {
+	transport transport
+	log       *FifoLog
+	cancel    context.CancelFunc
+}
+
+func (t *transportLog) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.cancel = cancel
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				entry, err := t.log.ReadLine()
+				if err != nil {
+					log.Error("Error on read log info", err)
+				}
+				if entry == nil {
+					time.Sleep(elapsedTime)
+				} else {
+					err = t.transport.WriteLog(entry)
+					if err != nil {
+						// cannot send, pushed again to the queue.
+						err := t.log.Write(entry)
+						if err != nil {
+							log.Error("cannot return log entry to the buffer log after transport failed", err)
+						}
+					}
+				}
+			}
+		}
+	}()
+}
+
+func (t *transportLog) Stop() {
+	if t.cancel != nil {
+		t.cancel()
+	}
+}
+
+type WorkloadsLogsTarget struct {
+	transports      map[string]transportLog
+	workloads       []*WorkloadWriter
+	workloadsConfig map[string]string
+	workloadManager *workload.WorkloadManager
+}
+
+func NewWorkloadsLogsTarget(manager *workload.WorkloadManager) *WorkloadsLogsTarget {
+	return &WorkloadsLogsTarget{
+		transports:      map[string]transportLog{},
+		workloads:       []*WorkloadWriter{},
```

Comment:
the name a bit confusing, since reading the line below `w.workloads[i].Stop()` caused me to think we stopped a running workload.
I think it should be renamed to `writers` (or go with full name `workloadWriters`)